synapse/storage/state.py

```diff
@@ -279,7 +289,7 @@ def _get_state_groups_from_groups_txn(self, txn, groups, types=None):
                     # after we finish deduping state, which requires this func)
                     args = [next_group]
                     if types:
```

Comment:
this can be unconditional now:

`args = [next_group] + where_args`

Reply:
yes, although unclear what that buys us? (plus in my lazyloading PR we do have other if's at that point)

Reply:
simplicity, mostly. but I agree it's not a big deal.